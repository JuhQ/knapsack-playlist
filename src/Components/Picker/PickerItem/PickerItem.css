.ui.items > .item.picker-selected {
  background-image: linear-gradient(
    128deg,
    var(--color-four),
    var(--color-one)
  );
}
.ui.items > .item.picker-selected .content {
  padding-top: 0.3rem;
}
.ui.items > .item.picker-selected .content .header,
.ui.items > .item.picker-selected .content .meta {
  font-family: Helvetica;
  font-weight: bold;
  font-size: 1.12rem;
  color: white;
  text-shadow: 0 0 8px #000;
}
.ui.items > .item.picker-selected:hover .content .header {
  color: var(--color-one);
}
