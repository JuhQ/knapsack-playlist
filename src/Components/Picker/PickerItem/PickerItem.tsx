import "./PickerItem.css";

import React, { useState } from "react";
import { Item } from "semantic-ui-react";

import { YoutubeItem } from "../../../types";
import Duration from "../../Duration/Duration";

interface Props {
  item: YoutubeItem;
  onClick: (item: YoutubeItem) => void;
}

const PickerItem: React.FC<Props> = ({ item, onClick }: Props) => {
  const [selected, setSelected] = useState<boolean>(false);

  return (
    <Item
      as="a"
      onClick={() => {
        setSelected(!selected);
        onClick(item);
      }}
      className={selected ? "picker-selected" : ""}
    >
      <Item.Image src={`https://i.ytimg.com/vi/${item.id}/hqdefault.jpg`} />
      <Item.Content>
        <Item.Header>{item.title}</Item.Header>
        <Item.Meta>
          <Duration seconds={item.seconds} />
        </Item.Meta>
      </Item.Content>
    </Item>
  );
};

export default PickerItem;
