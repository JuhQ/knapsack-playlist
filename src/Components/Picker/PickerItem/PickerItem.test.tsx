import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import React from "react";
import renderer from "react-test-renderer";

import PickerItem from "./PickerItem";

describe("PickerItem", () => {
  it("should render", () => {
    const component = renderer.create(
      <PickerItem
        item={{ id: "1", title: "test", seconds: 60, rating: 1 }}
        onClick={() => null}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be able to select item", () => {
    const handleClick = jest.fn();
    const wrapper = shallow(
      <PickerItem
        item={{ id: "2", title: "test 2", seconds: 60, rating: 1 }}
        onClick={handleClick}
      />
    );
    expect(handleClick).not.toHaveBeenCalled();
    wrapper.simulate("click");
    expect(handleClick).toHaveBeenCalled();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
