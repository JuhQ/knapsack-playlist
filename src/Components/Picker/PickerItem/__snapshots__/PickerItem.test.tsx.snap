// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerItem should be able to select item 1`] = `
<Item
  as="a"
  className="picker-selected"
  onClick={[Function]}
>
  <ItemImage
    src="https://i.ytimg.com/vi/2/hqdefault.jpg"
  />
  <ItemContent>
    <ItemHeader>
      test 2
    </ItemHeader>
    <ItemMeta>
      <Duration
        seconds={60}
      />
    </ItemMeta>
  </ItemContent>
</Item>
`;

exports[`PickerItem should render 1`] = `
<a
  className="item"
  onClick={[Function]}
>
  <div
    className="image"
  >
    <img
      src="https://i.ytimg.com/vi/1/hqdefault.jpg"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      test
    </div>
    <div
      className="meta"
    >
      1 minute
       
      
    </div>
  </div>
</a>
`;
